teradata/datadog_checks/teradata/config_models/validators.py

```diff
@@ -0,0 +1,38 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import ConfigurationError
+
+CREDENTIALS_OPTIONAL = ['JWT', 'KRB5', 'LDAP', 'TDNEGO']
+VALID_SSL_MODES = ['ALLOW', 'DISABLE', 'PREFER', 'REQUIRE']
+VALID_AUTH_MECHS = ['TD2', 'TDNEGO', 'LDAP', 'KRB5', 'JWT']
+AUTH_DATA_REQUIRED = ['JWT', 'KRB5', 'LDAP']
+
+
+def initialize_instance(values, **kwargs):
+    if values.get('auth_mechanism') and values.get('auth_mechanism').upper() not in CREDENTIALS_OPTIONAL:
+        if not values.get('username') or not values.get('password'):
+            raise ConfigurationError('`username` and `password` are required.')
```

Comment:
although this is a subclass, to be safe these should only raise ValueError, TypeError, or AssertionError https://pydantic-docs.helpmanual.io/usage/validators/